workers/unity/Packages/io.improbable.gdk.debug/MonoBehaviourInspector.cs

```diff
@@ -0,0 +1,192 @@
+                if (isWorkerType.HasValue)
+                {
+                    requiredWorkerTypesLabel = string.Join(" || ", requiredWorkerTypes);
+                }
+
+                var subscriptionSystem = linkedEntityComponent.World.GetExistingSystem<SubscriptionSystem>();
+
+                foreach (var fieldInfo in subscriptionsInfo.RequiredFields)
+                {
+                    var subscription =
+                        subscriptionSystem.Subscribe(linkedEntityComponent.EntityId, fieldInfo.FieldType);
+                    subscriptions.Add(fieldInfo, subscription);
+                }
+            }
+            else
+            {
+                if (requiredWorkerTypes != null)
+                {
+                    requiredWorkerTypesLabel = string.Join(" || ", requiredWorkerTypes);
+                }
+
+                foreach (var fieldInfo in subscriptionsInfo.RequiredFields)
+                {
+                    subscriptions.Add(fieldInfo, null);
+                }
+            }
+        }
+
+#if ODIN_INSPECTOR
+        protected override void OnDisable()
+#else
+        protected void OnDisable()
+#endif
+        {
+            if (subscriptions != null)
+            {
+                foreach (var pair in subscriptions)
+                {
+                    var subscription = pair.Value;
+                    subscription?.Cancel();
+                }
+            }
+        }
+
+        public override void OnInspectorGUI()
+        {
+            if (isSpatialBehaviour)
+            {
+                foldout = EditorGUILayout.Foldout(foldout, "SpatialOS", true);
+                if (foldout)
+                {
+                    using (new EditorGUI.IndentLevelScope())
+                    {
+                        if (Application.isPlaying && linkedEntityComponent != null)
+                        {
+                            DrawLiveInspector();
+                        }
+                        else
+                        {
+                            DrawEditorInspector();
+                        }
+                    }
+                }
+            }
+
+            base.OnInspectorGUI();
+        }
+
+        private void DrawEditorInspector()
+        {
+            if (!string.IsNullOrEmpty(requiredWorkerTypesLabel))
+            {
+                EditorGUILayout.LabelField(requiredWorkerTypesLabel);
+            }
+
+            if (subscriptions != null)
+            {
+                foreach (var pair in subscriptions)
+                {
+                    var field = pair.Key;
+                    EditorGUILayout.LabelField(field.FieldType.Name);
+                }
+            }
+        }
+
+        private void DrawLiveInspector()
+        {
+            if (isWorkerType.HasValue)
+            {
+                var style = isWorkerType.Value ? enabledContent : disabledContent;
+                style.text = isWorkerType.Value ? workerType : requiredWorkerTypesLabel;
```

Comment:
Nit: `style` is a little misleading imo

Reply:
`guiContent` ? As its of type `GUIContent`

Reply:
Sure 👍